commands/reminder.go

```diff
@@ -0,0 +1,91 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-redis/redis"
+)
+
+/*
+ * Set up redis table in a way where the keys are the dates of when to pull a reminder
+ * and the value is an array of events that need to be reminded on that date
+ */
+type Reminder struct {
+	Client *redis.Client
+}
+
+func NewReminder(url string) Reminder {
+	var (
+		storeurl string
+		client   *redis.Client
+	)
+	//For mock testing
+	if url != "" {
+		storeurl = url
+	} else {
+		//Replace with the config from the charts PR
+		storeurl = "localhost:6380"
+	}
+	client = redis.NewClient(&redis.Options{
+		Addr:     storeurl,
+		Password: "", //no password set
+		DB:       0,  //Use default DB
+	})
+	return Reminder{
+		Client: client,
+	}
+}
+
+//Add adds the new reminder as an entry into the redis table, we assume validation is
+//done when the message was recieved and the date should be in the form **/**/**
+//Append the reminder to the existing list if it exists, if not create a new list and add it
+func (r *Reminder) Add(message string, date string) error {
+	var (
+		sb  strings.Builder
+		val string
+	)
+	val, err := r.Client.Get(date).Result()
+	if err != nil {
```

Comment:
what other error messages are there? how come we don't return error from those

Reply:
the conditions make it so we only ignore one case `redis: nil` because redis returns an error on an empty value in a key, but I'm appending any new reminders to the same key so I need to make sure if a key has any values to begin with.

So if the value is nil, we ignore the error and start from an empty value